environments/jvm/src/main/java/io/fission/Server.java

```diff
@@ -59,10 +60,8 @@
 
 			jarFile = new JarFile(file);
 			Enumeration<JarEntry> e = jarFile.entries();
-			URL[] urls = { new URL("jar:file:" + file + "!/") };
+			URL[] urls = { file.toURI().toURL() };
 
-			// TODO Check if the classloading can be improved for ex. use something like:
-			// Thread.currentThread().setContextClassLoader(cl);
 			if (this.getClass().getClassLoader() == null) {
 				cl = URLClassLoader.newInstance(urls);
```

Comment:
I think we should look at this classloader example and see if we can do better here: https://stackoverflow.com/questions/60764/how-to-load-jar-files-dynamically-at-runtime/59743937#59743937

Reply:
Yes, I did read on that solution. The idea there is to have a custom class loader as the system class loader, which is more suited for a complex structure where you have to control loading of multiple classes. Also reading the cons - https://github.com/update4j/update4j/wiki/Documentation#cons-2 - I think there would be more work in this than the results we get.
Any thoughts are welcome if I am missing anything.

Reply:
Update4j contributed here, the cons listed are only relevant to the specific implementation of update4j's classloading model. Simply providing your own classloader as the system loader just works out of the box with the sole benefit of being able to cast `ClassLoader.getSystemClassLoader()` to your type.